packages/mds-audit/validators.ts

```diff
@@ -0,0 +1,102 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { TelemetrySchema } from '@mds-core/mds-ingest-service'
+import { providers } from '@mds-core/mds-providers'
+import { SchemaValidator } from '@mds-core/mds-schema-validators'
+import { AUDIT_EVENT_TYPES, Telemetry, VEHICLE_EVENTS } from '@mds-core/mds-types'
+import {
+  AuditApiAuditEndRequest,
+  AuditApiAuditNoteRequest,
+  AuditApiAuditStartRequest,
+  AuditApiVehicleEventRequest
+} from './types'
+
+const uuidSchema = { type: 'string', format: 'uuid' }
+const timestampSchema = { type: 'integer', minimum: 100_000_000_000, maximum: 99_999_999_999_999 }
+
+export const { isValid: isAuditApiAuditStartRequestValid } = SchemaValidator<AuditApiAuditStartRequest['body']>(
+  {
+    $id: 'AuditApiAuditStartRequest',
+    type: 'object',
+    properties: {
+      timestamp: timestampSchema,
+      provider_id: { ...uuidSchema, enum: Object.keys(providers) },
+      provider_vehicle_id: { type: 'string', maxLength: 255 },
+      audit_event_id: uuidSchema,
+      audit_device_id: uuidSchema,
+      telemetry: TelemetrySchema
+    },
+    required: ['timestamp', 'provider_vehicle_id', 'audit_event_id', 'audit_device_id']
+  },
+  { allErrors: true, removeAdditional: true }
+)
+
+export const { isValid: isValidUUID } = SchemaValidator<string>(uuidSchema)
+export const { isValid: isValidTimestamp } = SchemaValidator<number>(timestampSchema)
+
+export const { isValid: isValidAuditApiVehicleEventRequest } = SchemaValidator<AuditApiVehicleEventRequest['body']>({
+  $id: 'AuditApiVehicleEventRequest',
+  type: 'object',
+  properties: {
+    event_type: { type: 'string', enum: VEHICLE_EVENTS },
+    timestamp: timestampSchema,
+    telemetry: TelemetrySchema,
+    audit_event_id: uuidSchema
+  },
+  required: ['event_type', 'timestamp']
+})
+
+export const { isValid: isValidTelemetry } = SchemaValidator<Telemetry>(TelemetrySchema)
+
+export const { isValid: isValidAuditApiAuditNoteRequest } = SchemaValidator<AuditApiAuditNoteRequest['body']>({
+  $id: 'AuditApiAuditNoteRequest',
+  type: 'object',
+  properties: {
+    audit_event_id: uuidSchema,
+    audit_event_type: {
+      type: 'string',
+      enum: [AUDIT_EVENT_TYPES.issue, AUDIT_EVENT_TYPES.note, AUDIT_EVENT_TYPES.summary]
+    },
+    timestamp: timestampSchema,
+    telemetry: TelemetrySchema,
+    audit_issue_code: { type: 'string', maxLength: 31 },
+    note: { type: 'string', maxLength: 255 }
```

Comment:
Should probably be nullable:
```suggestion
    audit_issue_code: { type: 'string', maxLength: 31, nullable: true, default: null },
    note: { type: 'string', maxLength: 255, nullable: true, default: null }
```